packages/near-api-js/src/account.ts

```diff
@@ -710,4 +711,100 @@ export class Account {
             total: summary.total.toString(),
         };
     }
+
+    createTransaction(recevier: Account | string): Transaction {
+        return new Transaction(this, recevier);
+    }
+  
+    async hasDeployedContract(): Promise<boolean> {
+        return (await this.state()).code_hash !== EMPTY_CONTRACT_HASH;
+    }
 }
+
+
+/**
+ * Transaction Builder class. Initialized to an account that will sign the final transaction
+ */
+export class Transaction {
+    readonly receiverId: string;
+    readonly actions: Action[] = [];
+    private accountToBeCreated = false;
+    private _transferAmount?: BN;
+
+    constructor(private signer: Account, receiver: Account | string) {
```

Comment:
Given that we have an open issue around using named parameters, I'm thinking we should start writing new code using all named parameters.

Reply:
I think this is more for methods rather than constructors e.g. `Account` constructor args are `Connection` and `accountId`. I went for `Connection`, `senderId` and `receiverId` for `TransactionBuilder` since I felt `Connection` doesn't necessarily group well with the other args stylistically but maybe `Connection` and a `TransactionOptions` object with `senderId` and `receiverId`?

Reply:
>I think this is more for methods rather than constructors e.g. Account constructor args are Connection and 
accountId.

Ideally we'd change this for `Account` and `Connection` too 😄 

The goal behind named parameters is extendability and maintainability. They make it trivial to extend and refactor arguments and eliminate the [connascence of position](https://en.wikipedia.org/wiki/Connascence#Connascence_of_position_(CoP)) that makes methods unnecessarily brittle. JavaScript's handling of arguments can make positional arguments a liability, so having a single object argument with named parameters is not just a cosmetic concern.

The constructor benefits from this as much as any other method, and in fact it is a very common pattern in JavaScript development to have destructured object parameters in the constructor. Moreover having a single type representing dependencies for instantiating a class is convenient, consistent, and self-documenting.

> I went for Connection, senderId and receiverId for TransactionBuilder since I felt Connection doesn't necessarily group well with the other args stylistically but maybe Connection and a TransactionOptions object with senderId and receiverId?

There is no implication of grouping or association by including two arguments as keys on a single parameter object - again the benefit to naming your parameters is to make meaningful distinctions between them and enforce clarity in calling code. If you feel there is value to grouping parameters then it absolutely makes sense to do so within the same object, e.g.

```ts
type TransactionOptions = {
  receiverId: string;
  senderId: string;
};

type TransactionBuilderConfig = {
  connection: Connection;
  options: TransactionOptions;
};
...
constructor({ connection, options: { receiverId, senderId } }: TransactionBuilderConfig)
```

Personally I think it's a bit too verbose if the set of arguments is small and the names are clear, which in this case they are. I find grouping parameters is more useful when the set of arguments are more complex and where the arguments meanings benefit from additional context via partitioning.

Reply:
Makes sense, grouping into a `TransactionBuilderConfig` object with `connection`, `receiverId` and `senderId`